### DISCUSSION TITLE
API documentation for the repository - Auto-generated using Knowl CLI

### DISCUSSION THREAD

**Discussion by ashvin-sharma** [mathesar-foundation/mathesar#3440] [Show and tell]
Hello Everyone,

We have generated API documentation for the repository using Knowl CLI powered by generative AI.  This is generated for the entire repository in < 10 minutes and al lwithout relying on any annotations in code or any other developer effort. Also, we found it to be > 90% correct in our testing.

We are looking for feedback from the community on the documents and usability of these API docs in your day-to-day work 

- Access the documents here: [Knowl Generated Mathesar API Docs]( https://app.knowl.io/project/4c212ccd-4228-46b0-bf54-2e12cfd424c5 )
- More details are captured in the [blog](https://blog.knowl.ai/mathesar-knowl-generated-api-docs-3c917d7f3513) 

~Ashvin
github - ashvin-sharma
email: ashvin@knowl.io
discord: https://discord.gg/

**Comment by kgodey** [comment]
@ashvin-sharma We are still in alpha and our API is not stable, we're actually about to do a big refactor there. Unfortunately, we don't have the time to prioritize reviewing these auto-generated docs – they won't be useful for us. We don't use docs in our day to day workflow, and we won't be putting any effort into documenting our API for public use until we get to beta.
